shopinvader/services/abstract_sale.py

```diff
@@ -43,14 +46,28 @@ def _convert_step(self, sale):
     def _is_item(self, line):
         return True
 
-    def _convert_one_line(self, line):
+    def _get_product_to_convert(self, line):
+        """
+        Get the product to display on cart line
+        :param line:
+        :return: product.product
+        """
         if line.shopinvader_variant_id:
```

Comment:
the computed field is buggy. You could cherry pick this fix
https://github.com/shopinvader/odoo-shopinvader/pull/528/commits/617032d72b0512cf5cadecf522f3b8be1d6b9996

Reply:
Your fix would work only if you don't use guest mode.

Reply:
I think you should be interested by this one : https://github.com/shopinvader/odoo-shopinvader/pull/341

Reply:
@rousseldenis tnx for the reference! 
I have no time to check it in details but it seems that we should have a mix of both (my fix and the one from #341).
What do you think?

Reply:
@simahawk Could we merge this and then fix the field ?

Reply:
@sebastienbeau Waht do you think ?